src/core/exceptions.c

```diff
@@ -248,8 +248,8 @@ static MVMint32 search_frame_handlers_lex(MVMThreadContext *tc, MVMFrame *f,
             }
             if (skipping || !handler_can_handle(f, fh, cat, payload))
                 continue;
-            if (pc >= fh->start_offset &&
-                    pc <= fh->end_offset &&
+            if (((pc >= fh->start_offset && pc <= fh->end_offset) ||
+                    f->flags == MVM_FRAME_FLAG_EXIT_HAND_RUN) &&
```

Comment:
While you described the issue well in the commit message, I think this warrants also a comment in the code as it's quite a bit surprising.

Also what about the JITed case? Why don't we need any special handling there?

Reply:
My memory started to get fuzzy already, I'll have to look at this again.

Reply:
I've read into this again. I see no reason, why this shouldn't also affect JIT compiled frames. So, while I didn't manage to reproduce the error with only the non-jitted code path fixed, I've now added the respective fix to the jitted code path as well.

I've also added a respective comment to the code.